install/installer/pkg/components/openfga/deployment.go

```diff
@@ -137,8 +72,108 @@ func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
 				}},
 			},
 		}...)
+
+		// We use our cloud-sql-proxy sidecar to target the DB.
+		dbHost := "localhost"
+		openfgaEnvVars = append(openfgaEnvVars, []corev1.EnvVar{
+			{
+				Name:  "OPENFGA_DATASTORE_ENGINE",
+				Value: "mysql",
+			},
+			{
+				Name: "DB_PASSWORD",
+				ValueFrom: &corev1.EnvVarSource{SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: cfg.CloudSQL.DatabaseSecretRef,
+					},
+					Key: "password",
+				}},
+			},
+			{
+				Name: "DB_USERNAME",
+				ValueFrom: &corev1.EnvVarSource{SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: cfg.CloudSQL.DatabaseSecretRef,
+					},
+					Key: "user",
+				}},
+			},
+			{
+				Name:  "OPENFGA_DATASTORE_URI",
+				Value: fmt.Sprintf("$(DB_USERNAME):$(DB_PASSWORD)@tcp(%s:%d)/%s?parseTime=true", dbHost, CloudSQLProxyPort, cfg.CloudSQL.Instance),
+			},
+		}...)
+	}
```

Comment:
This is the core of this PRs change, rest are moves to satisfy declaration dependencies.